bpf/init.sh

```diff
@@ -656,7 +656,7 @@ if [ "$HOST_DEV1" != "$HOST_DEV2" ]; then
 fi
 
 # Remove bpf_xdp.o from previously used devices
-for iface in $(ip -o -a l | awk '{print $2}' | cut -d: -f1 | cut -d@ -f1 | grep -v cilium); do
+for iface in $(ip -o -a l | grep prog/xdp | awk '{print $2}' | cut -d: -f1 | cut -d@ -f1 | grep -v cilium); do
```

Comment:
@qmonnet IIRC, that `prog/xdp` tag isn't present with older versions of iproute2, right? Do you remember what version is needed?

Reply:
[Commit a0b5b7cf5c05](https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/commit/?id=a0b5b7cf5c05) (`bpf: consolidate dumps to use bpf_dump_prog_info`) seems to be in iproute2 v4.13.

Reply:
For what it's worth `bpftool net show` supports it from version 4.20, but with the added benefit of failing if listing XDP programs is not supported.

On an unrelated note, we have an occasion here to make the command above cleaner by doing the text processing in `awk` only instead of spawning additional processes for `cut` and `grep`. I would even be inclined to using `ip`'s JSON output, but I don't think we currently require `jq` be always present on the machines?

Reply:
Ok, so this change is safe to backport to 1.8 and 1.7 if needed since we use iproute2 v4.20 there.

Reply:
Hi, thanks for the comment. I think the newly added `grep` is expected to exclude most devices in the first place, so the existing `cut` and `grep` process may not affect performance, what about improving readability in a separate PR?